Dockerfile

```diff
@@ -0,0 +1,96 @@
+
+# Adds DeepX Gazebo Rendering image.
+#
+# Arguments:
+# - BASE_IMAGE: Base image.
+# - TAG: Tag of the BASE_IMAGE.
+#
+# Docker commands:
+# - To build the image:
+#   docker build --build-arg BASE_IMAGE=ghcr.io/deepx-inc/base_images --build-arg TAG=foxy -t deepx-gz-rendering .
+# - To publish the image:
+#   docker push ghcr.io/deepx-inc/deepx-gz-rendering
+
+
+ARG BASE_IMAGE
+ARG TAG
+ARG FROM_SOURCE
+FROM ${BASE_IMAGE}:${TAG} AS DEEPX_GZ_RENDERING
+
+# ---BUILD TOOLS---
+RUN apt -qq update \
+    && apt -qq install -y --no-install-recommends \
+    cmake \
+    ffmpeg \
+    freeglut3-dev \
+    g++-8 \
+    git \
+    gnupg \
+    libfreeimage-dev \
+    libglew-dev \
+    libogre-2.1-dev \
+    libxi-dev \
+    libxmu-dev \
+    ninja-build \
+    pkg-config \
+    redis-server \
+    redis-tools \
+    software-properties-common \
+    lsb-release \
+    wget \
+    curl
+
+# ---Temporary solution to stop flikering in gazebo
+# Before mesa version 22.0.2, there was no flickering issue.
+# After doing apt update && apt upgrade, the flickering appears.
+# This flickering is observed in mesa version 22.1.1
+# [TODO] Remove this apt-mark hold, after the issue is solved.
+RUN apt-mark hold \
+    libegl-mesa0 \
+    libegl1-mesa-dev \
+    libgl1-mesa-dev \
+    libgl1-mesa-dri \
+    libglapi-mesa \
+    libgles2-mesa-dev \
+    libglu1-mesa-dev \
+    libglu1-mesa \
+    libglx-mesa0 \
+    mesa-utils
+
+# ---STANDARD GAZEBO INSTALL---
+RUN apt -qq update && \
+    sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list' && \
+    sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-prerelease `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-prerelease.list' && \
+    wget http://packages.osrfoundation.org/gazebo.key -O - | apt-key add - && \
+    add-apt-repository ppa:kisak/kisak-mesa && \
+    curl -sSL http://get.gazebosim.org | sh && \
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 800 --slave /usr/bin/g++ g++ /usr/bin/g++-8 --slave /usr/bin/gcov gcov /usr/bin/gcov-8 && \
+    apt -qq update && \
+    apt -qq upgrade -y --no-install-recommends \
+    libignition-cmake2-dev \
+    libignition-common4-dev \
+    libignition-fuel-tools7-dev \
+    libignition-gazebo6-dev \
+    libignition-gui6-dev \
+    libignition-launch5-dev \
+    libignition-math6-dev \
+    libignition-msgs8-dev \
+    libignition-physics5-dev \
+    libignition-plugin-dev \
+    libignition-tools-dev \
+    libignition-transport11-dev \
+    libignition-utils-dev \
+    libsdformat12 \
+    libignition-plugin-dev \
+    && apt -qq -y autoclean \
+    && apt -qq -y autoremove \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR "/root/gazebo"
+
+COPY . gz-rendering/
+
+RUN mkdir -p gz-rendering/build && \
+    cd gz-rendering/build && \
+    cmake .. && \ 
+    make install
```

Comment:
We should change this `make install` to `cmake --build . --target install -j`. The `-j` argument instructs make to use all available cores, which significantly speeds up build time.

Reply:
Thank you! This is what I was looking for. As per my modification in the Dockerfile, I am removing this from the Dockerfile, so I will try to use it in the workflow instead.

Reply:
`-j` makes building very fast locally (on our server), but on Github for some reasons it's still about the same. Are we only allotted with 1 core? 🤔

Reply:
That may very well be the case. Also note that our local servers have over 40 cores available, so even if it's 2 or 4 cores, it's a huge difference.
You could try `make -j 16 install` or something along those lines and check if that speeds up execution (in case I/O is the bottleneck).

Reply:
Huh weird. The building process stopped, and actually it cannot be expanded. I will investigate.
<img width="513" alt="image" src="https://user-images.githubusercontent.com/104542776/196602203-283e94d7-3cfd-43df-bfb5-b523bdd1fff3.png">

Reply:
I observed that with `-j` or `-j 16`, it got stuck in `[ 67%] Building CXX object ogre/src/CMakeFiles/ignition-rendering6-ogre.dir/OgreLightVisual.cc.o`. Once refreshed, `Run tests` cannot be expanded anymore.
So I had to revert to non-parallel building, which is the status quo, and actually finished.
(I was already hopeful of the parallel build...)

Reply:
Weird. 仕方が無い。